### ISSUE TITLE
Root resource was present, but now absent.

### ISSUE THREAD

**Issue by nemrood** [terraform-providers/terraform-provider-azurerm#11782]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version
TF Version 0.14.5
AZ Version 2.45.1

### Affected Resource(s)

- azurerm_key_vault_secret
- random_string

### Terraform Configuration Files

Main.tf

```hcl
locals {
  tfsettings = yamldecode(file(var.setting))
}

module "tdf_az_key_vault_secret" {
  for_each       = { for vpnsite in local.tfsettings.vpnsites : vpnsite.name => vpnsite }
  source = "../../modules/tdf_az_key_vault_secret"
  key_vault_id = data.azurerm_key_vault.kv.id
  secret_name  = each.value.name 
  psk = random_string.rs[each.key].result
}

resource "random_string" "rs" {
    for_each       = { for vpnsite in local.tfsettings.vpnsites : vpnsite.name => vpnsite }
    length       = 14
    min_upper    = 2
    min_lower    = 2
    min_numeric  = 2
    min_special  = 2
}
```

Module.tf
```hcl
resource "azurerm_key_vault_secret" "psk" {
  name         = var.secret_name
  value        = var.psk
  key_vault_id = var.key_vault_id
}
```

### Debug Output

```yaml
module.tdf_az_key_vault_secret["TDFHB"].azurerm_key_vault_secret.psk: Still creating... [2m0s elapsed]
module.tdf_az_key_vault_secret["TS"].azurerm_key_vault_secret.psk: Still creating... [2m0s elapsed]
module.tdf_az_key_vault_secret["TAS"].azurerm_key_vault_secret.psk: Still creating... [2m0s elapsed]
module.tdf_az_key_vault_secret["EIP"].azurerm_key_vault_secret.psk: Still creating... [2m0s elapsed]
Error: Provider produced inconsistent result after apply
When applying changes to
module.tdf_az_key_vault_secret["TS"].azurerm_key_vault_secret.psk, provider
"registry.terraform.io/hashicorp/azurerm" produced an unexpected new value:
Root resource was present, but now absent.
This is a bug in the provider, which should be reported in the provider's own
issue tracker.
Error: Provider produced inconsistent result after apply
When applying changes to
module.tdf_az_key_vault_secret["EIP"].azurerm_key_vault_secret.psk, provider
"registry.terraform.io/hashicorp/azurerm" produced an unexpected new value:
Root resource was present, but now absent.
This is a bug in the provider, which should be reported in the provider's own
issue tracker.
Error: Provider produced inconsistent result after apply
When applying changes to
module.tdf_az_key_vault_secret["TDFHB"].azurerm_key_vault_secret.psk, provider
"registry.terraform.io/hashicorp/azurerm" produced an unexpected new value:
Root resource was present, but now absent.
This is a bug in the provider, which should be reported in the provider's own
issue tracker.
Error: Provider produced inconsistent result after apply
When applying changes to
module.tdf_az_key_vault_secret["TAS"].azurerm_key_vault_secret.psk, provider
"registry.terraform.io/hashicorp/azurerm" produced an unexpected new value:
Root resource was present, but now absent.
This is a bug in the provider, which should be reported in the provider's own
issue tracker.
```

### Panic Output

N/A

### Expected Behaviour

It shoud have  generated one random string for each vpnsite and create a secret on azure keyvault using that random string as a value and the vpnsite name as a name for the secret.

### Actual Behaviour

Check error above.

### Steps to Reproduce

Terraform Init
Terraform Validate
Terraform Plan
Terraform Apply

**Comment by magodo** [comment]
@nemrood Thank you for submitting this!

This issue should be fixed by #10477, which is released since v2.46.0. Would you please try to update your provider version and try again?

**Comment by tombuildsstuff** [comment]
Closing in favour of #11059
